### ISSUE TITLE
Livestream rewind / seekback limited to exactly 1:00:00 before live

### ISSUE THREAD

**Issue by cheintz** [TeamNewPipe/NewPipe#10158]
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.25.1

### Steps to reproduce the bug

1. Open a livestream on web that started at least an hour ago and allows seekback to the beginning (some apparently don't). 
2. Open that livestream in NewPipe
3. Attempt to seek to the beginning
4. Note that you can only seek back to exactly one hour prior to "live"

### Expected behavior

Can seek back to beginning of stream if possible for that stream (if allowed on web)

### Actual behavior

Seek back limited to exactly one hour before "live" (even if full seekback is possible on web)

### Affected Android/Custom ROM version

Android 13 / GrapheneOS TQ2B.230505.005.A1.2023060700

### Affected device model

Pixel 7 Pro

**Comment by AudricV** [comment]
Yes, that's a limitation of the HLS manifest returned to the iOS client, the one which has been found to return the highest duration. Supporting higher durations (up to 4 hours) may be added in the future with the newest versions of DASH manifests used as a fallback by YouTube clients if no adaptive streams are returned.

Writing a custom ExoPlayer `MediaSource` (and maybe more) to support manifest-less live media system of YouTube, system used by official clients for playback of running livestreams, may be also a possibility, but this would require a lot of time and custom code.

**Comment by dftf-stu** [comment]
Literally _every_ live video I've ever watched in NewPipe has _never_ allowed me to rewind them... I'm surprised you've been lucky-enough to find some this does work on. 🤔
